Assets/_Scripts/GameStatus.cs

```diff
@@ -3,36 +3,142 @@
 
 public class GameStatus : MonoBehaviour {
 
+	static public GameStatus Instance;
+
+
+	// LoseLife: You will die when you lose enough lives;
+	// GainScore: teamwork. Your death will increase the other team's Score
+	public enum GameMode{LoseLife, GainScore};
+	public GameMode gameMode;
+	// Target lives/scores in each mode for winning
+	public int GameTargetRounds = 2;
+	public GameObject playerPrefab;
+
 	// private int playerNum;
 	private bool isGameOver = false;
+	private int[] playerDeathCount = new int[4]{0,0,0,0};
+
+	private UserData[] userDataCollection = new UserData[4];
+
+	void Awake()
+	{
+		Instance = this;
+	}
 	// Use this for initialization
 	void Start () {
 		// playerNum = GameObject.FindGameObjectsWithTag (TagList.Player).Length;
+		BindPlayerData ();
+	}
+
+	void BindPlayerData()
+	{
+		GameObject[] playerCollection = GameObject.FindGameObjectsWithTag (TagList.Player);
+		foreach(GameObject player in playerCollection)
+		{
+			UserInputManager userCtrl = player.GetComponent<UserInputManager>();
+			int playerID = userCtrl.playerNum;
+			userDataCollection[playerID] = new UserData();
+			userDataCollection[playerID].userID = playerID;
+			userDataCollection[playerID].initPosition = player.transform.position;
+		}
 	}
 	
 	// Update is called once per frame
 	void Update () {
+
+	}
+
+	void WinEndGame()
+	{
 		GameObject[] playerCollection = GameObject.FindGameObjectsWithTag (TagList.Player);
-		if(playerCollection.Length <= 1)
+
+		isGameOver = true;
+
+		foreach(GameObject player in playerCollection)
 		{
-			if(playerCollection.Length == 1 && !isGameOver)
+			GameObject winEffPrefab = Resources.Load ("ArenaEffects/WinParEff") as GameObject;
+			GameObject winEff = GameObject.Instantiate (winEffPrefab, 
+			                                            player.transform.position, Quaternion.identity)
+				as GameObject;
+			UserInputManager userInput = player.GetComponent<UserInputManager>();
+			userInput.LockLeftInput(3.0f);
+		}
+		StartCoroutine(ReadyToRestart());
+	}
+
+	IEnumerator ReadyToRestart()
+	{
+		yield return new WaitForSeconds (3.0f);
+		Application.LoadLevel (Application.loadedLevel);
+	}
+
+	public void DecPlayerLife(int playerID)
```

Comment:
Dec is not meaningful.
Don't use any thing like especially in a public function.